### PR TITLE
[ModernBert] Prevent the attention mask from being None in ModernBertForSequenceClassification

### LINKED ISSUE #35917: Attention can be None in ModernBertForSequenceClassification
In the ModernBertForSequenceClassification class, the attention is never computed outside of the self.model (which is a ModernBertModel). Therefore when the attention is not used as input for the model the .unsqueeze() [here](https://github.com/huggingface/transformers/blob/86d7564611d21731fc004b4e79e472d48c4b0fec/src/transformers/models/modernbert/modular_modernbert.py#L1362) fails.
I solved this issue by assagning torch.ones(batch_size,seq_len) to the attention_mask, but I am not sure whether this is correct.

### PR DESCRIPTION
**Pull request by ashmikuz**
# What does this PR do?

This PR uses the same code for the attention mask in ModernBertForSequenceClassification as the one that is used inside the ModernBertModel class. This means that it will output a warning when the inputs contains padding but no attention_mask was specified, while also setting the attention mask to 1 to avoid it being None in later parts of the code (see #35917).
I did a quick test and it seems to fix the crash and print the warning when training on a sequence classification task.

Fixes #35917

### CODE REVIEW

**Comment by ashmikuz** [comment]
I had messed up and only committed the modeling file, not the modular, should be ok now.

**Comment by Rocketknight1** [comment]
Not sure who to ping for ModernBert - @tomaarsen, or @ArthurZucker for text models?

**Review by ArthurZucker — COMMENTED** [review]
Hey! Thanks but I am not sure I understand why we need this? It happens in `self.model`'s forward pass anyways?

**Comment by ashmikuz** [comment]
> Hey! Thanks but I am not sure I understand why we need this? It happens in `self.model`'s forward pass anyways?

The issue is that if the attention mask is None and input_ids is not the forward call will crash later, when attention_mask.unsqueeze() is called on None [here](https://github.com/huggingface/transformers/blob/86d7564611d21731fc004b4e79e472d48c4b0fec/src/transformers/models/modernbert/modular_modernbert.py#L1362). I know the code is redundant as it is called in the model as well, but this way the appropriate warning is shown if there is padding and no attention mask is provided, and attention_mask is never None.

**Review by ArthurZucker — COMMENTED** [review]
Ah... got it. It is very annoying to have to copy past the everywhere, but okay lets merge 🤗
